account/account.go

```diff
@@ -0,0 +1,69 @@
+package account
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	jsonpatch "github.com/evanphx/json-patch"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/pbsmetrics"
+	"github.com/prebid/prebid-server/stored_requests"
+)
+
+// GetAccount looks up the config.Account object referenced by the given accountID, with access rules applied
+func GetAccount(ctx context.Context, cfg *config.Configuration, fetcher stored_requests.AccountFetcher, accountID string) (account *config.Account, errs []error) {
+	// Check BlacklistedAcctMap until we have deprecated it
+	if _, found := cfg.BlacklistedAcctMap[accountID]; found {
+		return nil, []error{&errortypes.BlacklistedAcct{
+			Message: fmt.Sprintf("Prebid-server has disabled Account ID: %s, please reach out to the prebid server host.", accountID),
+		}}
+	}
+	if cfg.AccountRequired && accountID == pbsmetrics.PublisherUnknown {
+		return nil, []error{&errortypes.AcctRequired{
+			Message: fmt.Sprintf("Prebid-server has been configured to discard requests without a valid Account ID. Please reach out to the prebid server host."),
+		}}
+	}
+	if accountJSON, accErrs := fetcher.FetchAccount(ctx, accountID); len(accErrs) > 0 || accountJSON == nil {
+		// accountID does not reference a valid account
+		for _, e := range accErrs {
+			if _, ok := e.(stored_requests.NotFoundError); !ok {
+				errs = append(errs, e)
+			}
+		}
+		if cfg.AccountRequired && cfg.AccountDefaults.Disabled {
+			errs = append(errs, &errortypes.AcctRequired{
```

Comment:
If there are `FetchAccount` errors other than `NotFoundError`, then the error message here implies the publisher is sending an invalid account id where the host may be experiencing problems accessing it's accounts store. How do you feel about adding a differently worded message here when `accErrs` > 0? Perhaps something alongs the lines of:

Prebid-server could not verify the Account ID. Please reach out to the prebid server host.

Reply:
Sounds better. Best to avoid misleading messaging.
updated in cc4a174